open-scd.spec.ts

```diff
@@ -146,7 +163,197 @@ describe('with menu plugins loaded', () => {
     );
     await editor.updateComplete;
 
-    const plugin = editor.shadowRoot?.querySelector('*[docname="test.xml"]');
+    const plugin = editor.shadowRoot?.querySelector('*[docname="test.scd"]');
     expect(plugin).to.have.property('editCount', 1);
   });
 });
+
+describe('Custom plugins', () => {
+  beforeEach(async () => {
+    editor.plugins = {
+      menu: [testMenuPlugin],
+    };
+    await editor.updateComplete;
+
+    await waitUntil(
+      () =>
+        util.querySelectorWithTextContent(
+          editor.menuUI,
+          'mwc-list-item > span',
+          testMenuPlugin.name
+        ),
+      'Custom Plugin did not load'
+    );
+  });
+
+  it('executes the plugin upon menu item click', async () => {
+    const sclDoc = new DOMParser().parseFromString(
+      util.sclDocString,
+      'application/xml'
+    );
+    editor.dispatchEvent(newOpenEvent(sclDoc, 'test.scd'));
+    await editor.updateComplete;
+    const node = editor.doc.querySelector('Substation')!;
+    editor.dispatchEvent(newEditEvent({ node }));
+    await editor.updateComplete;
+    expect(sclDoc.querySelector('Substation')).to.not.exist;
+
+    editor.menuUI.open = true;
+    await editor.menuUI.updateComplete;
+    const pluginMenuItem = util
+      .querySelectorWithTextContent(
+        editor.menuUI,
+        'mwc-list-item > span',
+        'Test Undo Menu Plugin'
+      )
+      ?.closest('mwc-list-item');
+    expect(pluginMenuItem).to.exist;
+    expect(pluginMenuItem).to.have.property('disabled', false);
+    pluginMenuItem?.click();
+    await editor.updateComplete;
+    expect(sclDoc.querySelector('Substation')).to.exist;
+  });
+
+  it('does not allow the loading of a plugin, already loaded', async () => {
+    const currentEditorPluginCount = editor.loadedPlugins.menu.length;
+    editor.plugins = {
+      menu: [testMenuPlugin],
+      editor: [],
+    };
+    await editor.updateComplete;
+    expect(editor.loadedPlugins.menu.length).to.equal(currentEditorPluginCount);
```

Comment:
Are you sure that checking count is what we want here? In the next test it looks more natural to go by length, but here I'm not sure from looking just at the test that it doesn't have the same length of all the way through, regardless of whether it loads something.